src/core/components.py

```diff
@@ -448,20 +448,29 @@ class Modal(Component):
         "title",
         "components",
         "trigger_conditions",
+        "trigger"
     ]
 
-    def __init__(self, title, components, trigger_conditions=None, **kwargs):
+    def __init__(self, title, components, trigger_conditions=None, trigger=None, **kwargs):
         super().__init__(**kwargs)
         self.title = title
         self.components = components
         self.trigger_conditions = trigger_conditions or []
+        self.trigger = self.validate_trigger(trigger)
 
+    def validate_trigger(self, trigger):
+        trigger_values = ["onLoad", "onBlur", "onSubmit"]
+        if trigger in trigger_values:
+            return trigger
+        else:
+            raise InvalidArguments(f"Trigger must be either of the following: {', '.join(trigger_values)}")
```

Comment:
Nit: The error message reads a little odd to me, but it might just be me... :)

I'd suggest: "Trigger must be one of the following:"

Not a big deal at all, but may be worth checking this with someone else.